src/taipy/core/data/abstract_file.py

```diff
@@ -0,0 +1,101 @@
+# Copyright 2023 Avaiga Private Limited
+#
+# Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with
+# the License. You may obtain a copy of the License at
+#
+#        http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on
+# an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the
+# specific language governing permissions and limitations under the License.
+
+import os
+import re
+import urllib.parse
+from abc import abstractmethod
+from datetime import datetime, timedelta
+from typing import Dict, List, Optional, Set
+
+import modin.pandas as modin_pd
+import pandas as pd
+from sqlalchemy import create_engine, text
+
+from taipy.config.common.scope import Scope
+
+from .._version._version_manager_factory import _VersionManagerFactory
+from ..common.alias import DataNodeId, Edit
+from ..exceptions.exceptions import InvalidExposedType, MissingRequiredProperty, UnknownDatabaseEngine
+from .data_node import DataNode
+
+
+class _AbstractFileDataNode(DataNode):
+    """Abstract base class for data node implementations (CSVDataNode, ParquetDataNode, ExcelDataNode and
+    PickleDataNode) that are file based."""
+
+    __STORAGE_TYPE = "NOT_IMPLEMENTED"
+
+    __EXPOSED_TYPE_PROPERTY = "exposed_type"
+    __EXPOSED_TYPE_NUMPY = "numpy"
+    __EXPOSED_TYPE_PANDAS = "pandas"
+    __EXPOSED_TYPE_MODIN = "modin"
+    __VALID_STRING_EXPOSED_TYPES = [__EXPOSED_TYPE_PANDAS, __EXPOSED_TYPE_NUMPY, __EXPOSED_TYPE_MODIN]
+
+    __PATH_KEY = "path"
+    __DEFAULT_PATH_KEY = "default_path"
+    __HAS_HEADER_PROPERTY = "has_header"
+
+    _REQUIRED_PROPERTIES: List[str] = []
+
+    def __init__(
+        self,
+        config_id: str,
+        scope: Scope,
+        id: Optional[DataNodeId] = None,
+        name: Optional[str] = None,
+        owner_id: Optional[str] = None,
+        parent_ids: Optional[Set[str]] = None,
+        last_edit_date: Optional[datetime] = None,
+        edits: List[Edit] = None,
+        version: str = None,
+        validity_period: Optional[timedelta] = None,
+        edit_in_progress: bool = False,
+        properties: Dict = None,
+    ):
+        if properties is None:
+            properties = {}
+        if missing := set(self._REQUIRED_PROPERTIES) - set(properties.keys()):
+            raise MissingRequiredProperty(
+                f"The following properties " f"{', '.join(x for x in missing)} were not informed and are required"
+            )
```

Comment:
Actually this does not work, unfortunatelly, neither on the AbstractClass nor on Datanode. Because the _REQUIRED_PROPERTIES are diferent for each file data node, and we don't have a way for the parent class to actually know. We could implement a _check_required_properties method on either _AbstractFile or on Datanode and call it on the chield class, otherwise won't really work.

Reply:
Oh, yes... I would create a method in Datanode so it can be used in AbstractFile and other like SQLDataNOde, etc...

Reply:
So things have change in this case. I've noticed that the file type datanodes don't have any required properties anymore(The attribute is empty in all of the classes), the only information that would raise a MissingRequiredProperty is the `path`, which it won't anymore because we added the path generation when is not informed. For this reason, I just removed this check from all of the file base datanodes.

As far as SQL goals, it has a implementation on the AbstractSQL class that is already in use.